lib/rspec/support/object_inspector.rb

```diff
@@ -4,23 +4,42 @@ module Support
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
       # @api private
-      def self.inspect(object)
-        if Time === object
+      def self.inspect(initial)
+        formatted = formatter(initial)
+        if String === formatted || Symbol === formatted
```

Comment:
I just discovered another problem with this (while trying this branch against rspec/rspec-expectations#740, which updates that gem to use `RSpec::Support::ObjectInspector.inspect`).  `RSpec::Support::ObjectInspector.inspect(obj)` is intended to be usable in place of `obj.inspect`, but due to this conditional, it behaves differently for strings compared to `String#inspect`.  `String#inspect` prints double-quote marks around the string, where as your changes here cause the string to be printed without the double-quote marks since it skips the `inspect` call.  As a result, it causes many failures in rspec-expectations.